test/e2e/25_max_history.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+
+function setup() {
+  # Load libraries in setup() to access BATS_* variables 
+  load lib/env
+  load lib/install
+  load lib/poll
+
+  kubectl create namespace "$E2E_NAMESPACE"
+  install_git_srv
+  install_tiller
+  install_helm_operator_with_helm
+  kubectl create namespace "$DEMO_NAMESPACE"
+}
+
+@test "When max history is not set on release the most recent 10 revisions are kept" {
+  if [ "$HELM_VERSION" != "v3" ]; then
+    skip
+  fi
+
+  # Apply the HelmRelease
+  kubectl apply -f "$FIXTURES_DIR/releases/helm-repository.yaml" >&3
+  poll_until_equals 'podinfo-helm-repository HelmRelease' 'deployed' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.releaseStatus' --no-headers"
+
+  for i in {1..15}
+  do
+    # Apply a patch to initiate an upgrade
+    kubectl patch -f "$FIXTURES_DIR/releases/helm-repository.yaml" --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/values/someField\", \"value\":\"$i\"}]" >&3
+    poll_until_equals 'podinfo-helm-repository HelmRelease' "$(($i+1))" "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.observedGeneration' --no-headers"
+  done
+
+  # Check count of revisions is <= 10
+  count=$(k get secret -n $DEMO_NAMESPACE --output name | grep 'sh.helm.release.v1.podinfo-helm-repository' | wc -l)
+  [ "$count" -le 10 ]
+}
+
+@test "When max history on release is set to 5 the most recent 5 revisions are kept" {
+  if [ "$HELM_VERSION" != "v3" ]; then
+    skip
+  fi
+
+  # Apply the HelmRelease
+  kubectl apply -f "$FIXTURES_DIR/releases/helm-repository.yaml" >&3
+  poll_until_equals 'podinfo-helm-repository HelmRelease' 'deployed' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.releaseStatus' --no-headers"
+
+  # Patch HelmRelease to set maxHistory to 5
+  kubectl patch -f "$FIXTURES_DIR/releases/helm-repository.yaml" --type='json' -p='[{"op": "replace", "path": "/spec/maxHistory", "value":5}]' >&3
+  poll_until_equals 'podinfo-helm-repository HelmRelease' '2' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.observedGeneration' --no-headers"
+
+  for i in {1..8}
+  do
+    # Apply a patch to initiate an upgrade
+    kubectl patch -f "$FIXTURES_DIR/releases/helm-repository.yaml" --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/values/someField\", \"value\":\"$i\"}]" >&3
+    poll_until_equals 'podinfo-helm-repository HelmRelease' "$(($i+2))" "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.observedGeneration' --no-headers"
```

Comment:
What is the reason for `$(($i+2))` here instead of `$(($i+1))`?

Reply:
Reason being I apply the patch above to set the maxHistory so to start we are at revision 2 before we begin this loop.

Reply:
I think I wasn't fully awake yet this morning. Makes totally sense now.

Reply:
Is there a better way to wait for the rollout? it was all I could think of that was reliable.